lib/cjs/generated/rum.d.ts

```diff
@@ -725,6 +725,20 @@ export declare type RumViewEvent = CommonProperties & {
         js_refresh_rate?: RumPerfMetric;
         [k: string]: unknown;
     };
+    /**
+     * Parent view properties (view wrapping the current view)
+     */
+    readonly parent_view?: {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏On the replay side we have `nestedEnvId`. I think this is kind of the same concept, so maybe we could have a unified wording for both of those properties? like:

```
nested_env: {
   view: { id: string },
   source: ...
}
```

or `parent_env` (and we rename `nestedEnvId` to `parentEnvId` in replay)

Reply:
I don't have a strong opinion on this one but as we already have a parent hierarchy for RUM events (Session => view => error) maybe going all in for nested is better.

Reply:
I think it makes sense to have `parent_view` since we're adding it on the nested webview, while for replay `nested_env` is added on the wrapping wireframe (from the parent native view)

Reply:
> those new attributes on the RUM events would be exposed to customers.

Good point. Maybe they don't need to?

> while for replay nested_env is added on the wrapping wireframe

No it's are added to all Browser records too, not only the webview wireframe record: https://github.com/DataDog/rum-events-format/pull/165/files

Reply:
> > those new attributes on the RUM events would be exposed to customers.

> Good point. Maybe they don't need to?

Let's say that you want to see all errors generated on a specific page / screen regardless of the source, it would allow to search for `@view.id: xxx OR @parent_view.id:xxx`.
Not sure that we need to support this kind of use case, it could bring some value though.

Reply:
loosly held opinion: I prefer to avoid `nestedEnv` and stick with `parent_view.id` like `initial_view.id`

Reply:
Proposal: use "parent" (as "nested" is not quite clear)

* In *Browser Records*, rename `nestedEnvId` to `parentEnvId`

* In *Browser RUM events*:
  * rename `parent_view.id` to `parent.view.id`
  * rename `parent_view.source` to `parent.source`
  * (maybe one day we'll have `parent.env.id` or something like that if needed)

Reply:
I do agree with the first point (`nested` --> `parent`).
For the second point, I would say we can avoid nesting the id within an object. But if this proposal suits the rest, I can do the change.
What do you think @bcaudan @mquentin @ThibautGeriz

Reply:
> I do agree with the first point (nested --> parent).
Same here I feel it's a better naming overall.

`nestedEnvId` => `parentEnvId` works great to me then. (just remember that it will break all the current Mobile Webview example we have and are currently debuging)

considering that we already started with the `initial_view` sounds a good option to me `parent_view`. On top of that adding a later `parent_env` does not collide with `parent_view` as an env and a view will be different entities with different props

Reply:
> In Browser Records, rename nestedEnvId to parentEnvId

Do we also have mentions of nestedEnvId on some mobile records?
If yes, maybe we could remove the "nested" or "parent" prefix from the records.
FMU, the nested or parent aspect depends a bit on the point of view (from the parent or the child).

Reply:
> If yes, maybe we could remove the "nested" or "parent" prefix from all the records

This sounds good to me!

---

I see my proposal don't have much traction, so I'll write my thoughts and then feel free to merge. We are introducing a new customer facing concept here, and I think it's fine if we take a bit of time to find a good naming for it.

### Why I don't like having `view` in the newly introduced root namespace?

I proposed `parent.view.id` instead of `parent_view.id`, because we also want to add the parent `source` field, and the existing [`source` field](https://github.com/DataDog/rum-events-format/blob/d9624e702e86352babaf59979877f980eaccb8ff/lib/esm/generated/rum.d.ts#L891C5-L891C13) is a "root level" field (it's `@source`, not `@view.source`) so `@parent_view.source` does not make sense to me, and using a dedicated namespace to represent all the "parent environment" fields would be better. It would also allow to add more things unrelated to views into it, for example the "parent environment id".

I am aware of the `initial_view` property, and I agree that if we only had view-related fields, `parent_view` could be nice. But we don't.

### Maybe we need another name for that concept?

Maxime [noticed](https://github.com/DataDog/rum-events-format/pull/170#discussion_r1394524194) that the "parent" concept is already taken for the relationship between Session -> View -> Action -> Error / Resource / Longtask. It is customer facing as we are mentioning "Parent hierarchy" in the UI.

If we were introducing those new fields only on View events, it could have been fine, it could be seen as a new layer of the "parent hierarchy".

But those fields will be on all events. So a customer might wonder which field to use to target the parent view of an Error event: should they use `@view.id` or `@parent_view.id`?

Now, I understand `nested_env` is not quite clear. I proposed `parent_env`, but maybe we can think of something else?

Reply:
> We are introducing a new customer facing concept here, and I think it's fine if we take a bit of time to find a good naming for it.

👍 super interesting discussion

> I proposed parent.view.id instead of parent_view.id, because we also want to add the parent source field

👍 That makes sense to me too

> the "parent" concept is already taken for the relationship between Session -> View -> Action -> Error / Resource / Longtask.

What about something like "container" or "host" instead of parent then?

Other thought: 
if we plan to expose the "env" concept on the RUM side, it would also collides with the existing one (service/env/version) 😕
Renaming it to something like "runtime" could work.

So it could look like
- mobile webview record: runtimeId to identify the id of the corresponding webview
- browser record: runtimeId to identify the id of the webview they are in
- browser rum events: container_runtime.* to attach data from the mobile side (view, source)